torchvision/transforms/v2/functional/_color.py

```diff
@@ -670,3 +671,45 @@ def invert(inpt: datapoints._InputTypeJIT) -> datapoints._InputTypeJIT:
             f"Input can either be a plain tensor, any TorchVision datapoint, or a PIL image, "
             f"but got {type(inpt)} instead."
         )
+
+
+def permute_channels_image_tensor(image: torch.Tensor, permutation: List[int]) -> torch.Tensor:
+    shape = image.shape
+    num_channels, height, width = shape[-3:]
+
+    if len(permutation) != num_channels:
+        raise ValueError(
+            f"Length of permutation does not match number of channels: " f"{len(permutation)} != {num_channels}"
+        )
+
+    if image.numel() == 0:
+        return image
+
+    image = image.reshape(-1, num_channels, height, width)
```

Comment:
should this just be `image = image[..., permutation, :, :]`? maybe this could simplify the code

Reply:
Are JIT considerations still relevant? From my discussions in core, I understood that it is now deprecated, and that Dynamo is the relevant technology. Do you mean that it is also not supported by Dynamo?

Reply:
> From my discussions in core, I understood that it is now deprecated

That would be nice, but so far it is not true. torchscript has not been officially deprecated yet. Hence we still need to keep support for it.

So far I have only seen chatter from individual maintainers that said as much, but my current understanding is that dynamo is not yet a full replacement. Meaning, until that is the case, we can't drop torchscript support.

As soon as there is an official deprecation notice, we can also act on that. Depending on the status of v2, we might go through a deprecation cycle or drop it sooner.